miio/integrations/vacuum/mijia/g1vacuum.py

```diff
@@ -373,3 +373,18 @@ def consumable_reset(self, consumable: G1Consumable):
     def set_fan_speed(self, fan_speed: G1FanSpeed):
         """Set fan speed."""
         return self.set_property("fan_speed", fan_speed.value)
+
+    @command()
+    def fan_speed_presets(self) -> FanspeedPresets:
+        """Retrieve predefined fan speeds, see VacuumInterface for details."""
```

Comment:
I'm not sure if the custom click group handler works with properties, but I think it would look nicer if this would be a property (to indicate that it does no I/O).

Reply:
That's not true. On one device, the implementation depends on version of the device. So it is necessary to detect version first.

Reply:
Yes, but we are moving towards that direction anyway when #1328 or something similar gets introduced. In my mind, the instances would be constructed by the lib based on information from `miIO.info` call, so this information would be available. Anyway, this can also wait until we are at that point.